xmovie/core.py

```diff
@@ -358,12 +370,66 @@ def save_frames(self, odir, progress=False):
         elif ~tqdm_avail and progress:
             warnings.warn("Cant show progess bar at this point. Install tqdm")
 
-        for fi in frame_range:
-            fig, ax, pp = self.render_frame(fi)
-            frame_save(
-                fig, fi, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
+        for timestep in frame_range:
+            fig, ax, pp = self.render_single_frame(timestep)
+            save_single_frame(
+                fig, timestep, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
             )
 
+
+
+    def save_frames_parallel(self, odir, progress=False, 
+                             parallel_compute_kwargs=dict()):
+        '''
+        Saves all frames in parallel using dask.map_blocks.
+
+        Parameters
+        ----------
+        odir : path
+            path to output directory
+        progress : Bool
+            Whether or not to show a progress bar.
+        parallel_compute_kwargs : dict
+            Keyword arguments to pass t dask's `compute()` function.
+
+        '''
+        import numpy as np
+        import dask.array as darray
+
+        da = self.data
+        framedim = self.framedim
+        
+        da = da.chunk({self.framedim : 1}) # DataArray needs to have length-1 chunks in framedim
+        chunk_dims = [ dim for dim in da.dims if dim != framedim ] # Create dimensions for each chunk
+               
+
+        total_time = da[framedim]
+        def create_render_save_single_frame2(xr_array, framedim):
+            time_of_chunk = xr_array[framedim]
+            timestep = abs(total_time - time_of_chunk[0]).argmin().item() # get index of chunk in framedim
+
+            fig, ax, pp = self.render_single_frame(timestep)
+            save_single_frame(fig, timestep, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi)
+
+            return time_of_chunk
+
+        mapped_save_and_close_frames = da.map_blocks(func=create_render_save_single_frame2,
```

Comment:
Is `mapped_save_and_close_frames` used somewhere else? Or can we just call the line and not assign the output.

Reply:
It's only used there. I prefer that way because it looks more organized and I think it doesn't incur any significant slowdowns. But I can change if you think it's better to just call the line directly.

Reply:
I would prefer to remove it, just because it might confuse another user (it did confuse me a little).